### PR TITLE
chore(object store): change streaming write buffer size to 16MB for gcs

### PR DESCRIPTION
**Pull request by wcy-fdu**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Review by Xuanwo — APPROVED** [review]
Thanks!

File: src/object_store/src/object/opendal_engine/gcs.rs [hunk 1]
```diff
@@ -28,6 +30,8 @@ impl OpendalObjectStore {
 
         builder.root(&root);
 
+        builder.write_fixed_size(GCS_BUFFER_SIZE);
```
Comment (Li0k) [inline src/object_store/src/object/opendal_engine/gcs.rs:33]: any context or benchmark ?

Comment (hzxa21) [reply src/object_store/src/object/opendal_engine/gcs.rs:33]: We noticed that the part size of streaming upload for GCS is default to 8MB in the opendal implementation, which is different from the part size (16MB) we are uing for AWS S3. We decided to keep them as the same for consistency.

File: src/object_store/src/object/opendal_engine/gcs.rs [hunk 2]
```diff
@@ -18,6 +18,8 @@ use opendal::Operator;
 
 use super::{EngineType, OpendalObjectStore};
 use crate::object::ObjectResult;
+
+const GCS_BUFFER_SIZE: usize = 16 * 1024 * 1024;
```
Comment (hzxa21) [inline src/object_store/src/object/opendal_engine/gcs.rs:22]: Can you add some documentation that this is the part size used in streaming upload?

**Review by hzxa21 — APPROVED** [review]
LGTM.

I noticed that OpenDAL exposes [similar configurations ](https://docs.rs/opendal/latest/opendal/layers/index.html?search=write_fixed_size) for OSS and COS. How about also making them consistent with AWS and GCS?

**Comment by Xuanwo** [comment]
> I noticed that OpenDAL exposes [similar configurations ](https://docs.rs/opendal/latest/opendal/layers/index.html?search=write_fixed_size) for OSS and COS. How about also making them consistent with AWS and GCS?

I have a plan to support something like:

```rust
let w: Writer = op.writer_with(path).buffer_size(16 * 1024 * 1024).await?;
```

So that we can configure the buffer size at runtime.

**Comment by wcy-fdu** [comment]
> How about also making them consistent with AWS and GCS?

Sure, let me apply this change on OSS (COS is not implemented via OpenDAL but s3 compatible).

File: src/object_store/src/object/opendal_engine/gcs.rs (hunk 1 above)
Comment (wcy-fdu) [reply src/object_store/src/object/opendal_engine/gcs.rs:33]: yes, just to keep consistency, you can refer to https://cloud.google.com/storage/pricing?hl=zh-cn for the price.

**Comment by wcy-fdu** [comment]
> > I noticed that OpenDAL exposes [similar configurations ](https://docs.rs/opendal/latest/opendal/layers/index.html?search=write_fixed_size) for OSS and COS. How about also making them consistent with AWS and GCS?
> 
> I have a plan to support something like:
> 
> ```rust
> let w: Writer = op.writer_with(path).buffer_size(16 * 1024 * 1024).await?;
> ```
> 
> So that we can configure the buffer size at runtime.

Go ahead~

**Pull request merged**